plugin/etcd/plugin.go

```diff
@@ -0,0 +1,451 @@
+				Title:   "Full Overwrite",
+				Example: "false",
+			},
+			plugin.Field{
+				Mode:    "target",
+				Name:    "prefix",
+				Type:    "string",
+				Help:    "This is the input string for prefix based backup-restore",
+				Title:   "Prefix",
+				Example: "starkandwayne/",
+			},
+		},
+	}
+	fmt.Fprintf(os.Stderr, "etcd plugin starting up...\n")
+	plugin.Run(p)
+}
+
+type EtcdPlugin plugin.PluginInfo
+
+type EtcdConfig struct {
+	EtcdEndpoints  string
+	Authentication bool
+	Username       string
+	Password       string
+	ClientCertPath string
+	ClientKeyPath  string
+	CaCertPath     string
+	FullOverwrite  bool
+	Prefix         string
+}
+
+func (p EtcdPlugin) Meta() plugin.PluginInfo {
+	return plugin.PluginInfo(p)
+}
+
+func getEtcdConfig(endpoint plugin.ShieldEndpoint) (*EtcdConfig, error) {
+	etcdEndpoint, err := endpoint.StringValue("endpoints")
+	if err != nil {
+		return nil, err
+	}
+
+	auth, err := endpoint.BooleanValueDefault("auth", false)
+	if err != nil {
+		return nil, err
+	}
+
+	user, err := endpoint.StringValueDefault("user", "")
+	if err != nil {
+		return nil, err
+	}
+
+	pass, err := endpoint.StringValueDefault("pass", "")
+	if err != nil {
+		return nil, err
+	}
+
+	clientCert, err := endpoint.StringValueDefault("clientCertPath", "")
+	if err != nil {
+		return nil, err
+	}
+
+	clientKey, err := endpoint.StringValueDefault("clientKeyPath", "")
+	if err != nil {
+		return nil, err
+	}
+	caCert, err := endpoint.StringValueDefault("caCertPath", "")
+	if err != nil {
+		return nil, err
+	}
+
+	fullOverwrite, err := endpoint.BooleanValueDefault("fullOverwrite", false)
+	if err != nil {
+		return nil, err
+	}
+
+	prefix, err := endpoint.StringValueDefault("prefix", "")
+	if err != nil {
+		return nil, err
+	}
+
+	return &EtcdConfig{
+		EtcdEndpoints:  etcdEndpoint,
+		Authentication: auth,
+		Username:       user,
+		Password:       pass,
+		ClientCertPath: clientCert,
+		ClientKeyPath:  clientKey,
+		CaCertPath:     caCert,
+		FullOverwrite:  fullOverwrite,
+		Prefix:         prefix,
+	}, nil
+}
+
+func (p EtcdPlugin) Validate(endpoint plugin.ShieldEndpoint) error {
+	var (
+		s    string
+		b    bool
+		err  error
+		fail bool
+	)
+	s, err = endpoint.StringValue("endpoints")
+	if err != nil {
+		fmt.Printf("@R{\u2717 endpoints  %s}\n", err)
+		fail = true
+	} else {
+		fmt.Printf("@G{\u2713 etcd endpoints} data in @C{%s} will be backed up\n", s)
+	}
+
+	b, err = endpoint.BooleanValueDefault("auth", false)
+	if err != nil {
+		fmt.Printf("@R{\u2717 authentication %s}\n", err)
+		fail = true
+	} else if b {
+		fmt.Printf("@G{\u2713 auth} authentication is enabled\n")
+	} else {
+		fmt.Printf("@G{\u2713 auth} authentication is disabled\n")
+	}
+
+	if b {
+		s, err = endpoint.StringValueDefault("user", "")
+		if err != nil {
+			fmt.Printf("@R{\u2717 user  %s}\n", err)
+			fail = true
+		} else if s == "" {
+			fmt.Printf("@G{\u2713 user} username was not provided so cert based auth will be used\n")
```

Comment:
This is probably a bad idea; change `auth` to a string / enum, and be more explicit about value management.  I.e. if auth is `x509`, do cert stuff, if it's `none`, don't do auth, etc.